fastapi/_compat.py

```diff
@@ -109,9 +110,22 @@ def type_(self) -> Any:
             return self.field_info.annotation
 
         def __post_init__(self) -> None:
-            self._type_adapter: TypeAdapter[Any] = TypeAdapter(
-                Annotated[self.field_info.annotation, self.field_info]
-            )
+            with warnings.catch_warnings():
+                # Pydantic >= 2.12.0 warns about field specific metadata that is unused
+                # (e.g. `TypeAdapter(Annotated[int, Field(alias='b')])`). In some cases, we
+                # end up building the type adapter from a model field annotation so we
+                # need to ignore the warning:
+                if PYDANTIC_VERSION_MINOR_TUPLE >= (2, 12):  # pragma: no cover
```

Comment:
Once Pydantic 2.12 is released, we'll be able to remove the `no cover` statement here.

Reply:
(Which should happen very soon, hopefully today).

Reply:
(I know, but we have a bit of a chicken-or-egg situation going on here 😉)